clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -805,16 +820,32 @@
         RESPONSES[ApiKeys.DESCRIBE_GROUPS.id] = DESCRIBE_GROUPS_RESPONSE;
         RESPONSES[ApiKeys.LIST_GROUPS.id] = LIST_GROUPS_RESPONSE;
         RESPONSES[ApiKeys.SASL_HANDSHAKE.id] = SASL_HANDSHAKE_RESPONSE;
+        RESPONSES[ApiKeys.API_VERSIONS.id] = API_VERSIONS_RESPONSE;
 
-        /* set the maximum version of each api */
-        for (ApiKeys api : ApiKeys.values())
+        /* set the minimum and maximum version of each api */
+        for (ApiKeys api : ApiKeys.values()) {
             CURR_VERSION[api.id] = (short) (REQUESTS[api.id].length - 1);
+            for (int i = 0; i < REQUESTS[api.id].length; ++i)
+                if (REQUESTS[api.id][i] != null) {
+                    MIN_VERSIONS[api.id] = (short) i;
+                    break;
+                }
+        }
 
-        /* sanity check that we have the same number of request and response versions for each api */
-        for (ApiKeys api : ApiKeys.values())
+        /* sanity check that:
+             - we have the same number of request and response versions for each api
+             - for each supported api, minimum supported version is same for its request and response*/
```

Comment:
I said this elsewhere, but the comment was hidden because the diff changed. It seems to me that we are not just checking the minimum, consider if there's a null element after a non-null element in the array. We are checking that we have a consistent set of request and response versions.

Also for multi-line comments, we generally include a `*` on each starting line.